src/higherorderfns.jl

```diff
@@ -156,14 +156,18 @@ map!(f::Tf, C::AbstractSparseMatrixCSC, A::AbstractSparseMatrixCSC, Bs::Vararg{S
 map!(f::Tf, C::SparseVecOrMat, A::SparseVecOrMat, Bs::Vararg{SparseVecOrMat,N}) where {Tf,N} =
     (_checksameshape(C, A, Bs...); _noshapecheck_map!(f, C, A, Bs...))
 function _noshapecheck_map!(f::Tf, C::SparseVecOrMat, A::SparseVecOrMat, Bs::Vararg{SparseVecOrMat,N}) where {Tf,N}
-    fofzeros = f(_zeros_eltypes(A, Bs...)...)
-    fpreszeros = _iszero(fofzeros)
+    haszeros = _haszeros(C) && _haszeros(A) && all(_haszeros, Bs)
+    # Avoid calculating f(zero) unless necessary (it may fail)
+    fofzeros = haszeros ? f(_zeros_eltypes(A, Bs...)...) : nothing
+    fpreszeros = !haszeros || _iszero(fofzeros)
```

Comment:
```suggestion
   !haszero || _iszero(f(_zeros_eltypes(A, Bs...)...))
```
what you wrote is type unstable.